tests/variable_creation_test.py

```diff
@@ -647,6 +647,27 @@ def test_arange_with_variables_mixed_dtype():
         sc.array(dims=['x'], values=[1, 2, 3], dtype='int64'))
 
 
+def test_arange_with_int32_creates_int32_array():
+    assert sc.identical(sc.arange('x', np.int32(2)),
+                        sc.array(dims=['x'], values=[0, 1], dtype='int32'))
+    assert sc.identical(sc.arange('x', np.int32(2), np.int32(4)),
+                        sc.array(dims=['x'], values=[2, 3], dtype='int32'))
+    assert sc.identical(sc.arange('x', np.int32(2), np.int32(4), np.int32(2)),
+                        sc.array(dims=['x'], values=[2], dtype='int32'))
+
+
+def test_arange_with_int32_variables_creates_int32_array():
```

Comment:
Should an `arange` created from a `float32` make a Variable of dtype `float32`?
And if so, we should add tests for that.